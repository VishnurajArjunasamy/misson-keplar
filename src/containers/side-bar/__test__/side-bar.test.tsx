import { render, screen, fireEvent } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import "@testing-library/jest-dom";
import SideBar from "../side-bar";
import sideBarReducer, {
  toggleDarkMode,
  toggleShowMembers,
} from "../../../store/side-bar-slice";

// Mocking components
jest.mock("../../filter-section/filter-section", () => () => (
  <div>FilterSection</div>
));
// jest.mock("../../components/modal/modal", () => ({ children, closeModal }) => (
//   <div onClick={closeModal}>
//     <div>{children}</div>
//   </div>
// ));
jest.mock("../../members-section/members-section", () => () => (
  <div>MembersSection</div>
));

// jest.mock("../../../store/side-bar-slice", () => ({
//     toggleDarkMode: jest.fn(),
//   }));

let store = configureStore({
  reducer: { sideBar: sideBarReducer },
  preloadedState: {
    sideBar: {
      showMembers: false,
      isDarkMode: false,
      members: null,
      loading: false,
      error: null,
    },
  },
});

interface UpdatedValues {
  showMembers?: boolean;
  isDarkMode?: boolean;
  members?: any;
  loading?: boolean;
  error?: any;
}

function updateStore(updatedValues: UpdatedValues) {
  return configureStore({
    reducer: { sideBar: sideBarReducer },
    preloadedState: {
      sideBar: {
        ...store.getState().sideBar,
        ...updatedValues,
      },
    },
  });
}

describe("SideBar Component", () => {
  beforeEach(() => {
  });

  it("renders the component correctly", () => {
    render(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );

    expect(screen.getByText("Little")).toBeInTheDocument();
    expect(screen.getByText("FilterSection")).toBeInTheDocument();
  });

  it("dispatches toggleShowMembers when 'View Members' is clicked", () => {
    render(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );

    const mockDispatch = jest.fn(); 
    fireEvent.click(screen.getByText("View Members"));
  });

  it("dispatches toggleDarkMode when 'Switch Dark Mode' is clicked", () => {
    render(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );
    const mockDispatch=jest.fn()
    const darkModeButton = screen.getByText("Switch to Dark Mode");
    fireEvent.click(darkModeButton);
  });

  it("displays the modal when showMembers is true", () => {
    store = updateStore({ showMembers: true });

    render(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );

    expect(screen.getByText("MembersSection")).toBeInTheDocument();
  });

  it("toggles dark mode class based on isDarkMode", () => {
    store = updateStore({ isDarkMode: true });

    render(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );

    expect(screen.getByText("FilterSection").parentElement).toHaveClass("dark");

    //     store = updateStore({ isDarkMode: false });

    //     render(
    //       <Provider store={store}>
    //         <SideBar />
    //       </Provider>
    //     );

    //     expect(
    //       screen.getByText("FilterSection").parentElement?.parentElement
    //     ).toHaveClass("light");
  });
});
function mockStore(arg0: {
  sideBar: { isDarkMode: boolean; showMembers: boolean };
}): import("@reduxjs/toolkit").EnhancedStore<
  { sideBar: import("../../../modals/sidebar-modal").SideBarState },
  import("redux").UnknownAction,
  import("@reduxjs/toolkit").Tuple<
    [
      import("redux").StoreEnhancer<{
        dispatch: import("redux-thunk").ThunkDispatch<
          { sideBar: import("../../../modals/sidebar-modal").SideBarState },
          undefined,
          import("redux").UnknownAction
        >;
      }>,
      import("redux").StoreEnhancer
    ]
  >
> {
  throw new Error("Function not implemented.");
}
