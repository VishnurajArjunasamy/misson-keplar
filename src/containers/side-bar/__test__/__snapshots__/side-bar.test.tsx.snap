// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component macthes snapshot 1`] = `
<DocumentFragment>
  <section
    class="sideBar dark"
  >
    <h1
      class="heading"
    >
      <span>
        Little
      </span>
      <span>
        Book
      </span>
    </h1>
    <div>
      FilterSection
    </div>
    <div
      class="actions"
    >
      <h2>
        View Members
      </h2>
      <h2>
        Switch to Light Mode
      </h2>
    </div>
    <div
      class="modal dark"
    >
      <div
        class="content slideIn "
      >
        <div>
          MembersSection
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
